### PR TITLE
compatible  with v2

### PR DESCRIPTION
**Pull request by feilin-flexcompute**

### CODE REVIEW

File: flow360client/case.py [hunk 1]
```diff
@@ -117,41 +154,50 @@ def GetCaseSurfaceForces(caseId, surfaces):
...
+
     try:
-        obj = s3Client.get_object(Bucket=Config.CASE_BUCKET,
-                                  Key='users/{0}/{1}/results/{2}'.format(keys['UserId'], caseId, 'surface_forces.csv'))
-        data = obj['Body'].read().decode('utf-8')
-    except Exception as e:
-        print('no surface forces available')
-        return None
+        obj = s3Client.get_object(Bucket=Config.CASE_BUCKET, Key=objKey2)
+    except:
```
Comment (chaitan3) [inline flow360client/case.py:171]: Abstract line 166 to line 176 into a function as it is reused in multiple places.

File: flow360client/case.py [hunk 2]
```diff
@@ -168,14 +214,14 @@ def GetCaseSummationOfSurfacesForces(caseId, surfaces):
 
         forceTable = forcesOnPatches[list(forcesOnPatches.keys())[0]]
         for forceType, forceData in forceTable.items():
-            if forceType == "steps":
+            if forceType == "steps" or forceType == "physical_step" or forceType == "pseudo_step":
```
Comment (chaitan3) [inline flow360client/case.py:217]: Easier code for this:
`if forceType in ["steps", "physical_step", "pseudo_step"]:`

File: flow360client/case.py [hunk 3]
```diff
@@ -168,14 +214,14 @@ def GetCaseSummationOfSurfacesForces(caseId, surfaces):
...
                 forcesCombo[surfaceCombo][forceType] = forceData
             else:
                 forcesCombo[surfaceCombo][forceType] = [0.0]*len(forceData)
 
         for patchName, forceTable in forcesOnPatches.items():
             for forceType, forceData in forceTable.items():
-                if forceType != "steps":
+                if forceType != "steps" and forceType != "physical_step" and forceType != "pseudo_step":
```
Comment (chaitan3) [inline flow360client/case.py:224]: Easier code for this:
`if not forceType in ["steps", "physical_step", "pseudo_step"]:`

Comment (feilin-flexcompute) [reply flow360client/case.py]: good tick

File: flow360client/case.py (hunk 2 above)
Comment (feilin-flexcompute) [reply flow360client/case.py]: updated. good trick

File: flow360client/case.py (hunk 1 above)
Comment (feilin-flexcompute) [reply flow360client/case.py]: add a new recursive function getS3Obj()

File: flow360client/case.py [hunk 4]
```diff
@@ -65,31 +66,62 @@ def ListCases(name=None, status=None, meshId=None, include_deleted=False):
...
-                                  Key='users/{0}/{1}/results/{2}'.format(keys['UserId'], caseId, 'surface_forces.csv'))
-        data = obj['Body'].read().decode('utf-8')
-    except Exception as e:
-        print('no surface forces available')
-        return None
+        obj = s3Client.get_object(Bucket=Config.CASE_BUCKET, Key=objKey2)
+    except:
+        try:
+            obj = s3Client.get_object(Bucket=Config.CASE_BUCKET, Key=objKey1)
+        except Exception as e:
```
[the remark this answers is not in the document]
Comment (feilin-flexcompute) [reply flow360client/case.py]: updated

File: flow360client/case.py [hunk 5]
```diff
@@ -65,31 +66,62 @@ def ListCases(name=None, status=None, meshId=None, include_deleted=False):
...
+
     try:
-        obj = s3Client.get_object(Bucket=Config.CASE_BUCKET,
-                                  Key='users/{0}/{1}/results/{2}'.format(keys['UserId'], caseId, 'surface_forces.csv'))
-        data = obj['Body'].read().decode('utf-8')
-    except Exception as e:
-        print('no surface forces available')
-        return None
+        obj = s3Client.get_object(Bucket=Config.CASE_BUCKET, Key=objKey2)
+    except:
```
[the remark this answers is not in the document]
Comment (feilin-flexcompute) [reply flow360client/case.py]: updated

File: flow360client/case.py [hunk 6]
```diff
@@ -65,31 +66,62 @@ def ListCases(name=None, status=None, meshId=None, include_deleted=False):
...
+        return resp
 
 @refreshToken
 def GetCaseTotalForces(caseId):
-    url = f'case/{caseId}/totalForces'
-    resp = get2(url)
+    try:
+        resp = get2(f'case/{caseId}/result/total_forces_v2')
+        return resp
+    except:
```
[the remark this answers is not in the document]
Comment (feilin-flexcompute) [reply flow360client/case.py]: updated

File: flow360client/case.py [hunk 7]
```diff
@@ -65,31 +66,62 @@ def ListCases(name=None, status=None, meshId=None, include_deleted=False):
...
 @refreshToken
 def GetCaseResidual(caseId):
-
-    url = f'case/{caseId}/residual'
-    resp = get2(url)
-    return resp
+    try:
+        resp = get2(f'case/{caseId}/result/nonlinear_residual_v2')
+        return resp
+    except:
```
[the remark this answers is not in the document]
Comment (feilin-flexcompute) [reply flow360client/case.py]: updated

File: flow360client/case.py [hunk 8]
```diff
@@ -1,3 +1,4 @@
+import pdb
```
[the remark this answers is not in the document]
Comment (feilin-flexcompute) [reply flow360client/case.py]: done

File: flow360client/case.py [hunk 9]
```diff
@@ -65,31 +65,68 @@ def ListCases(name=None, status=None, meshId=None, include_deleted=False):
...
+    except s3Client.exceptions.NoSuchKey as e:
+        return getS3Obj(keys[1:])
 
-    headerKeys = ['steps',
-                  'CL', 'CD', 'CFx', 'CFy', 'CFz', 'CMx', 'CMy', 'CMz',
-                  'CLPressure', 'CDPressure', 'CFxPressure', 'CFyPressure', 'CFzPressure', 'CMxPressure', 'CMyPressure', 'CMzPressure',
-                  'CLViscous', 'CDViscous', 'CFxViscous', 'CFyViscous', 'CFzViscous', 'CMxViscous', 'CMyViscous', 'CMzViscous']
+@refreshToken
+def GetCaseSurfaceForces(caseId, surfaces):
+    objKey1 = 'users/{0}/{1}/results/{2}'.format(keys['UserId'], caseId, 'surface_forces.csv')
```
Comment (chaitan3) [inline flow360client/case.py:112]: Line 112 to Line 129 is repeated at Line 172 to 187. We should unify the whole block into a single helper function

Comment (feilin-flexcompute) [reply flow360client/case.py:112]: updated. I created a new function call getStepAndForceKeys()

**Review by chaitan3 — APPROVED** [review]

**Review by chaitan3 — APPROVED** [review]

**Pull request merged**